source/solvers/cahn_hilliard.cc

```diff
@@ -39,52 +40,210 @@ CahnHilliard<dim>::setup_assemblers()
         std::make_shared<CahnHilliardAssemblerBDF<dim>>(
           this->simulation_control));
     }
-  // Core assembler
+
+  // Angle of contact boundary condition
   this->assemblers.push_back(
-    std::make_shared<CahnHilliardAssemblerCore<dim>>(this->simulation_control));
+    std::make_shared<CahnHilliardAssemblerAngleOfContact<dim>>(
+      this->simulation_control,
+      this->simulation_parameters.multiphysics.ch_parameters,
+      this->simulation_parameters.boundary_conditions_cahn_hilliard));
+
+
+  // Core assembler
+  this->assemblers.push_back(std::make_shared<CahnHilliardAssemblerCore<dim>>(
+    this->simulation_control,
+    this->simulation_parameters.multiphysics.ch_parameters));
 }
 
 template <int dim>
 void
 CahnHilliard<dim>::assemble_system_matrix()
 {
-  return;
+  this->system_matrix = 0;
+  setup_assemblers();
+
+  const DoFHandler<dim> *dof_handler_fluid =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+
+  auto scratch_data = CahnHilliardScratchData<dim>(
+    this->simulation_parameters.physical_properties_manager,
+    *this->fe,
+    *this->cell_quadrature,
+    *this->mapping,
+    dof_handler_fluid->get_fe(),
+    *this->face_quadrature);
+
+  WorkStream::run(this->dof_handler.begin_active(),
+                  this->dof_handler.end(),
+                  *this,
+                  &CahnHilliard::assemble_local_system_matrix,
+                  &CahnHilliard::copy_local_matrix_to_global_matrix,
+                  scratch_data,
+                  StabilizedMethodsCopyData(this->fe->n_dofs_per_cell(),
+                                            this->cell_quadrature->size()));
+
+  system_matrix.compress(VectorOperation::add);
 }
 
 template <int dim>
 void
-CahnHilliard<dim>::assemble_local_system_matrix()
+CahnHilliard<dim>::assemble_local_system_matrix(
+  const typename DoFHandler<dim>::active_cell_iterator &cell,
+  CahnHilliardScratchData<dim> &                        scratch_data,
+  StabilizedMethodsCopyData &                           copy_data)
 {
-  return;
+  copy_data.cell_is_local = cell->is_locally_owned();
+  if (!cell->is_locally_owned())
```

Comment:
```suggestion
  if (!copy_data.cell_is_local)
```

Reply:
why not put everything directly inside the if?